lib/Redis.php

```diff
@@ -275,37 +275,29 @@ private function send (array $strings, callable $responseCallback = null, $addFu
 				if ($error) {
 					$promisor->fail($error);
 				} else {
-					$promisor->succeed($this->doSend($strings, $responseCallback, $addFuture));
-				}
-			});
-		} else {
-			$promisor = new Failure(new ConnectException("connection has already been closed"));
-		}
-
-		return $promisor;
-	}
-
-	private function doSend (array $strings, callable $responseCallback = null, $addFuture = true) {
-		$payload = "";
+		                        $payload = "";
+		                        foreach ($strings as $string) {
+			                	$payload .= sprintf("$%d\r\n%s\r\n", strlen($string), $string);
+					}
+					$payload = sprintf("*%d\r\n%s", sizeof($strings), $payload);
 
-		foreach ($strings as $string) {
-			$payload .= sprintf("$%d\r\n%s\r\n", strlen($string), $string);
-		}
+					$future = null;
+					if ($addFuture) {
+						$future = new RedisFuture($responseCallback);
```

Comment:
Shouldnt we use a regular future when `$responseCallback` is null? This would save us from invoking `RedisFuture->succeed` just for the sake of calling `parent->succeed` their without further logic.